imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,187 @@
+            try:
+                self._image_file = rawpy.imread(request.get_file())
+            except rawpy.NotSupportedError:
+                if request._uri_type == URI_BYTES:
+                    raise InitializationError(
+                        "RawPy can not read the provided bytes."
+                    ) from None
+                else:
+                    raise InitializationError(
+                        f"RawPy can not read {request.raw_uri}."
+                    ) from None
+        elif request.mode.io_mode == IOMode.write:
+            raise InitializationError(
+                "RawPy does not support writing."
+            ) from None
+
+    def close(self) -> None:
+
+        if self._image_file:
+            self._image_file.close()
+
+        self._request.finish()
+
+    def read(self, *, index: int = 0, **kwargs) -> np.ndarray:
+        """Read Raw Image.
+
+        Returns
+        -------
+        nd_image: ndarray
+            The image data
+        """
+
+        nd_image: np.ndarray
+
+        try:
+            nd_image = self._image_file.postprocess(**kwargs)
+        except Exception as ex:
+            raise ex
+
+        return nd_image
+
+    def write(
+        self, 
+        ndimage: Union[ArrayLike, List[ArrayLike]]
+    ) -> bytes | None:
+        """RawPy does not support writing.
+        """
+        raise NotImplementedError()
+
+    def iter(self) -> Iterator[np.ndarray]:
+        """Load the image.
+
+        Returns
+        -------
+        nd_image: ndarray
+            The image data
+        """
+
+        try:
+            yield self.read()
+        except Exception as ex:
+            raise ex
+
+    def metadata(
+        self, index: int = None, exclude_applied: bool = True
+    ) -> Dict[str, Any]:
+        """Read ndimage metadata.
+
+        Parameters
+        ----------
+        exclude_applied : bool
+            If True, exclude metadata fields that are applied to the image while
+            reading. For example, if the binary data contains a rotation flag,
+            the image is rotated by default and the rotation flag is excluded
+            from the metadata to avoid confusion.
+
+        Returns
+        -------
+        metadata : dict
+            A dictionary of format-specific metadata.
+
+        """
+
+        metadata = {}
+
+        ImageSize = self._image_file.sizes
+
+        # TO DO: Find out what information needs to be included below and refine accordingly.
+        metadata["black_level_per_channel"] = self._image_file.black_level_per_channel
+        metadata["camera_white_level_per_channel"] = self._image_file.camera_white_level_per_channel
+        metadata["color_desc"] = self._image_file.color_desc
+        #metadata["color_matrix"] = self._image_file.color_matrix
+        metadata["daylight_whitebalance"] = self._image_file.daylight_whitebalance
+        metadata["dtype"] = self._image_file.raw_image.dtype.type
```

Comment:
```suggestion
        metadata["dtype"] = self._image_file.raw_image.dtype
```

Shouldn't this just be the numpy `dtype`?

Reply:
I tried using only dtype in the beginning but I was getting some error. I tried again today and now it looks like it is working. May be the error was happening because of something else. I have updated the file now.